Floating App Bar Recipe

brianegan opened the pull request:
A recipe describing how to use the CustomScrollView and SliverAppBar together to create a floating app bar!

sfshaza2 commented on src/docs/cookbook/lists/floating-app-bar.md:
```diff
@@ -0,0 +1,196 @@
+displays a "tall" app bar that occupies a lot of vertical space.
+
+Traditionally, you create an app bar by providing an `appBar` property to the
+`Scaffold` Widget. This will create a fixed app bar that always remains above
```
will create => creates

brianegan replied:
Ah, I just can't stop myself from making this mistake, haha :)

sfshaza2 commented on src/docs/cookbook/lists/floating-app-bar.md:
```diff
@@ -0,0 +1,196 @@
+`Scaffold` Widget. This will create a fixed app bar that always remains above
+the `body` of the `Scaffold`.
+
+If you'd like to create a floating app bar scrolls offscreen as the user scrolls
```
floating app bar scrolls => floating app bar that scrolls .  ??

sfshaza2 commented on src/docs/cookbook/lists/floating-app-bar.md:
```diff
@@ -0,0 +1,196 @@
+through a list, you need to move the app bar from the Scaffold into a
+[`CustomScrollView`](https://docs.flutter.io/flutter/widgets/CustomScrollView-class.html).
+This allows you to create an app bar that scrolls offscreen as you scroll
+through a list of items contained inside the `CustomScrollView`!
```
This feels a little repetitive with the first sentence. How about:

Moving the app bar from the Scaffold into the [`CustomScrollView`][] allows you to create an
app bar that scrolls offscreen as you scroll through a list of items contained inside the
`CustomScrollView`.

sfshaza2 commented on src/docs/cookbook/lists/floating-app-bar.md:
```diff
@@ -0,0 +1,196 @@
+
+The scrollable lists and widgets that can be provided to the the
+`CustomScrollView` are known as slivers. There are several types of Slivers,
+such as a `SliverList`, `SliverGridList`, and `SliverAppBar`! In fact, the
```
Global. Replace the exclamation points with periods. :)

sfshaza2 commented on src/docs/cookbook/lists/floating-app-bar.md:
```diff
@@ -0,0 +1,196 @@
+Play around with the [various properties you can pass to the `SliverAppBar`
+Widget](https://docs.flutter.io/flutter/material/SliverAppBar/SliverAppBar.html)
+and use hot reload to see the results. For example, you can use an `Image`
+Widget for the `flexibleSpace` to create a background image that shrinks in size
```
for the `flexibleSpacd` to => for the `flexibleSpace` property to

sfshaza2 commented on src/docs/cookbook/lists/floating-app-bar.md:
```diff
@@ -0,0 +1,196 @@
+The `SliverList` and `SliverGrid` Widgets take one required parameter: a
+[`SliverChildDelegate`](https://docs.flutter.io/flutter/widgets/SliverChildDelegate-class.html).
+While this sounds fancy, the delegate is simply used to provide a list Widgets
+to to `SliverList` or `SliverGrid`! For example, the
```
to to => to

sfshaza2 commented on src/docs/cookbook/lists/floating-app-bar.md:
```diff
@@ -0,0 +1,196 @@
+While this sounds fancy, the delegate is simply used to provide a list Widgets
+to to `SliverList` or `SliverGrid`! For example, the
+[`SliverChildBuilderDelegate`](https://docs.flutter.io/flutter/widgets/SliverChildBuilderDelegate-class.html)
+allows you to create a list of items that are built lazily as your scroll, just
```
your scroll => you scroll

brianegan commented:
Thanks for the review, @Sfshaza :) I've addressed all feedback. Please let me know if it looks good or if ya see anything else that needs improvement.
